build: MUSL and eglibc provide these symbols

pprindeville opened the pull request:
build: Bind needs setresuid() to build properly in recent versions.

cc: @nbd168 @dangowrt @jow-

pprindeville committed:
build: autoconf: set ac_cv_func_setresuid=true

MUSL and eglibc provide these symbols.

Bind needs setresuid() to build properly in recent versions.

Signed-off-by: Philip Prindeville <philipp@redfish-solutions.com>
Link: https://github.com/openwrt/openwrt/pull/15952
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

pprindeville commented:
bind-9.19.24 won't build with the current cache.

pprindeville commented:
Blocks: https://github.com/openwrt/packages/pull/24594

kempniu commented on include/site/linux:
```diff
@@ -37,7 +37,7 @@ ac_cv_func_rindex=yes
 ac_cv_func_setgrent_void=yes
 ac_cv_func_setpgrp_void=yes
-ac_cv_func_setresuid=no
+ac_cv_func_setresuid=yes
```
IMHO it would be better to remove this line altogether than change its value.  Removing it would allow Autoconf to perform its checks as it normally would for any other `AC_CHECK_FUNCS()` call.

Forcing any value in this site script, be it `no` or `yes`, prevents a package `Makefile` from overriding it.  Removing it from this site script allows a package `Makefile` to set the value of this variable as it desires, if that ever becomes necessary to do explicitly.

AIUI, `setresuid()` should be treated identically as its sibling functions, e.g. `setresgid()`, which do not have any Autoconf cache values forced here.

pprindeville replied:
That raises the question, "Do we need this file at all?"

hauke replied:
I would also suggest to remove it here.

pprindeville replied:
Remote the entire file in this PR?

hauke replied:
Setting it to true should be fine.

hauke commented on include/site/linux:
```diff
@@ -66,9 +66,12 @@ ac_cv_sctp=no
 ac_cv_time_r_type=POSIX
 ac_cv_type_suseconds_t=yes
+ac_cv_size_t=yes
+ac_cv_ssize_t=yes
```
Why are you adding these?

pprindeville replied:
Aren't they present in all platforms?

hauke replied:
yes I think so. If you want to add them please do it in a separate commit with a description in the commit message.

pprindeville committed:
build: autoconf: Activate options for standard types

These are POSIX standard types

And therefore should be present in any POSIX compliant runtime.

Signed-off-by: Philip Prindeville <philipp@redfish-solutions.com>
Link: https://github.com/openwrt/openwrt/pull/15952
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>
